onnxruntime/python/tools/quantization/qdq_quantizer.py

```diff
@@ -27,14 +27,28 @@
 
 class QDQQuantizer(ONNXQuantizer):
     def __init__(self, model, per_channel, reduce_range, mode, static, weight_qType, input_qType, tensors_range,
-                 nodes_to_quantize, nodes_to_exclude, op_types_to_quantize, extra_options={}):
+                 nodes_to_quantize, nodes_to_exclude, op_types_to_quantize, op_types_to_exclude_output_quantization=[], extra_options={}):
         ONNXQuantizer.__init__(self, model, per_channel, reduce_range, mode, static, weight_qType, input_qType,
                                tensors_range, nodes_to_quantize, nodes_to_exclude, op_types_to_quantize, extra_options)
         self.tensors_to_quantize = []
         self.tensors_to_quantize_per_channel = []
         self.bias_to_quantize = []
         self.nodes_to_remove = []
 
+        # Specific op types to exclude qdq quantization for their outputs.
+        # In some cases, for example QDQ BERT model for TensorRT, 
+        # adding QDQ for node's output may end up with worse accuracy.
+        # So, we don't recommend to add QDQ to node's output under such condition.
+        self.op_types_to_exclude_output_quantization = op_types_to_exclude_output_quantization
+
+        # In some cases, for example QDQ BERT model for TensorRT,
+        # QDQ should always appear as a pair. 
+        # For our quantization tool, we do quantization on Dequantizelinear's input 
+        # to remove Quantizelinear as optimization for weight.   
+        # Therefore, we need to disable this optimization and add qdq pair to weight.
+        self.add_qdq_pair_to_weight = False if 'AddQDQPairToWeight' not in extra_options \
```

Comment:
add those options to the extra_options

Reply:
What do you mean by this? 
I think it's already in extra_options, as extra_options['AddQDQPairToWeight']

Reply:
You need to put it down in the quantize.py

Reply:
[https://github.com/microsoft/onnxruntime/blob/3e5dbfd94f30ee328f3fc6b305f37d10e44a1d85/onnxruntime/python/tools/quantization/quantize.py#L182](https://github.com/microsoft/onnxruntime/blob/3e5dbfd94f30ee328f3fc6b305f37d10e44a1d85/onnxruntime/python/tools/quantization/quantize.py#L182)